service-statistics-sotka/src/main/java/fi/nls/oskari/control/statistics/plugins/sotka/SotkaIndicatorValuesFetcher.java

```diff
@@ -6,105 +6,131 @@
+                                           String regionCategoryId) {
+        Map<Integer, IndicatorValue> allValues = fetchData(selectors, indicator);
+        Map<String, IndicatorValue> filteredValues = new HashMap<>();
+        for (Entry<Integer, IndicatorValue> entry : allValues.entrySet()) {
+            Integer regionId = entry.getKey();
+            if (regionId == null) {
+                // sometimes this is null with 0.0 as value.
+                // TODO: check if this is a bug in some other code or an empty row from sotkanet
+                continue;
+            }
+
+            IndicatorValue value = entry.getValue();
+            // SotkaNET gives "Kunta" in some places, "KUNTA" in others...
+            String category = regionParser.getCategoryById(regionId);
+            if (category != null && category.equalsIgnoreCase(regionCategoryId)) {
+                // We must include this value to the result.
+                // Mapping sotka regionId to region id of layer!
+                String code = regionParser.getCode(regionId);
+                if (code != null) {
+                    filteredValues.put(code, value);
+                }
+            }
+        }
+        return filteredValues;
+    }
+
     /**
      * This returns the indicator data for all the layers, for every region category, from "maakunta" to "kunta", etc.
      * Note: Indexed by id, because id is unique, code is not.
+     *
      * @param selectors
      * @param indicator
      * @return
      */
-    public Map<Integer, IndicatorValue> getAll(StatisticalIndicatorDataModel selectors, String indicator) {
-        SotkaRequest request = SotkaRequest.getInstance(IndicatorData.NAME);
+    protected Map<Integer, IndicatorValue> fetchData(StatisticalIndicatorDataModel selectors, String indicator) {
+        SotkaRequest request = SotkaRequest.getInstance(IndicatorDataJSON.NAME);
         request.setBaseURL(config.getUrl());
         // If there is no defined values for gender or year, we will use "total" and an empty list.
         String gender = "total";
         List<String> years = new ArrayList<>();
         for (StatisticalIndicatorDataDimension selector : selectors.getDimensions()) {
-            switch(selector.getId()) {
-            case "sex":
-                gender = selector.getValue();
-                break;
-            case "year":
-                // Even though SotkaNET API supports fetching multiple years, we don't do that here.
-                // Multiple years can be fetched over several requests.
-                years.add(selector.getValue());
-                break;
-            default:
+            switch (selector.getId()) {
+                case "sex":
+                    gender = selector.getValue();
+                    break;
+                case "year":
+                    // Even though SotkaNET API supports fetching multiple years, we don't do that here.
```

Comment:
What does this comment mean? You are collecting multiple year values and using them in `IndicatorDataJSON.getRequestSpecificParams())`